crates/red_knot/src/module.rs

```diff
@@ -908,4 +969,66 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn relative_name() -> std::io::Result<()> {
+        let TestCase {
+            src,
+            db,
+            temp_dir: _temp_dir,
+            ..
+        } = create_resolver()?;
+
+        let foo_dir = src.path().join("foo");
+        let foo_path = foo_dir.join("__init__.py");
+        let bar_path = foo_dir.join("bar.py");
+
+        std::fs::create_dir(&foo_dir)?;
+        std::fs::write(foo_path, "from .bar import test")?;
+        std::fs::write(bar_path, "test = 'Hello world'")?;
+
+        let foo_module = db.resolve_module(ModuleName::new("foo")).unwrap();
+        let bar_module = db.resolve_module(ModuleName::new("foo.bar")).unwrap();
+
+        // `from . import bar` in `foo/__init__.py` resolves to `foo`
+        assert_eq!(
+            Some(ModuleName::new("foo")),
+            foo_module.relative_name(&db, 1, None)
+        );
+
+        // `from baz import bar` in `foo/__init__.py` should resolve to `foo/baz.py`
+        assert_eq!(
+            Some(ModuleName::new("foo.baz")),
+            foo_module.relative_name(&db, 0, Some("baz"))
+        );
```

Comment:
This one isn't right, `from baz import bar` is not a relative import (no dots), it's an absolute import, so this should resolve to just `baz`.

Either `relative_import` should error on `level = 0`, or it should just resolve it as an absolute import.